ch_backup/logic/lock_manager.py

```diff
@@ -106,7 +108,23 @@ def _zk_flock(self) -> None:
             if not client.exists(self._process_zk_lockfile_path):
                 client.create(self._process_zk_lockfile_path, makepath=True)
             self._zk_lock = client.Lock(self._process_zk_lockfile_path)
-            if not self._zk_lock.acquire(blocking=False):
-                sys.exit(self._exitcode)
+            force_exit = self._exitcode == 0
+            try:
+                if not self._zk_lock.acquire(blocking=True, timeout=self._lock_timeout):
+                    fail_on_lock_acquiring(force_exit)
+            except LockTimeout as e:
+                fail_on_lock_acquiring(force_exit, e)
         else:
             raise RuntimeError("ZK flock enabled, but zookeeper is not configured")
+
+
+def fail_on_lock_acquiring(force_exit: bool, e: Optional[Exception] = None) -> None:
+    """
+    Determines how to exit from the lock acquisition process
+    """
+    if force_exit:
+        sys.exit(0)
+    err = RuntimeError("Lock was not acquired")
+    if e is not None:
+        raise err from e
+    raise err
```

Comment:
Does it handled somewhere ? How about 
- Just `sys.exit(self._exitcode)` here or
- Handling `lock_config.exitcode` where we catch and handle this exception to respect `lock_conf.exitcode` in all cases